nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/Lock.java

```diff
@@ -36,10 +36,13 @@
 
     private final boolean failed;
 
+    private final int cachedHashCode;
+
     public Lock(String owner, Calendar created, boolean failed) {
         this.owner = owner;
         this.created = created;
         this.failed = failed;
+        this.cachedHashCode = calculateHashCode();
```

Comment:
No need to pre-calculate the hash code. Here, I would say that having a field to carry this value is more memory consumed regarding the benefit of saved CPU ressource.

Reply:
The cached prop is a primitive int. If you ever use these locks in hash collections this will make a significant difference.
I let you decide !

Reply:
Optimization you're talking about may be true for collections with a really huge amount of objects. Question is: do we have such cases for `Lock`? In the end this is premature optimization, and we tend to not do that. You can remove it.

Reply:
In that case we can also omit implementing hashCode as it would only be useful in such cases. WDYT

Reply:
Your call, on my side I implement the three methods `equals`/`hashCode`/`toString` or none of them.

Reply:
I think it's safer. I'll add the toString as well !